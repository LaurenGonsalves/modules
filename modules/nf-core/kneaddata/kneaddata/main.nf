process KNEADDATA_KNEADDATA {
    tag "${meta.id}"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/kneaddata:0.12.0--pyhdfd78af_1':
        'biocontainers/kneaddata:0.12.0--pyhdfd78af_1' }"

    input:
    tuple val(meta), path(fastq_gz)
    path (kneaddata_db)

    output:
    tuple val(meta), path("${prefix}_kneaddata_paired_{1,2}.fastq.gz")  , emit: preprocessed_reads
    tuple val(meta), path("${prefix}_kneaddata.log")                    , emit: kneaddata_log
    path "versions.yml"                                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    # install trimmomatic
    curl \\
    -o Trimmomatic-0.33.zip \\
    http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.33.zip

    # unzip trimmomatic executable
    unzip Trimmomatic-0.33.zip

    kneaddata \\
        --input1 ${fastq_gz[0]} \\
        --input2 ${fastq_gz[1]} \\
        --output ./ \\
        --output-prefix ${prefix}_kneaddata \\
        --reference-db ${kneaddata_db} \\
        --threads ${task.cpus} \\
        --trimmomatic ./Trimmomatic-0.33 \\
        $args

    gzip *.fastq

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        kneaddata: \$(echo \$(kneaddata --version 2>&1 | sed 's/^.*v//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_kneaddata_paired_1.fastq
    touch ${prefix}_kneaddata_paired_2.fastq
    gzip ${prefix}_kneaddata_paired_*.fastq
    touch ${prefix}_kneaddata.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        kneaddata: \$(echo \$(kneaddata --version 2>&1 | sed 's/^.*v//' ))
    END_VERSIONS
    """
}
