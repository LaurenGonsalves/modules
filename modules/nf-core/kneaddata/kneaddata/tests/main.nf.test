nextflow_process {

    name "Test Process: KNEADDATA_KNEADDATA"
    script "../main.nf"
    process "KNEADDATA_KNEADDATA"

    tag "modules"
    tag "modules_local"
    tag "kneaddata"
    tag "kneaddata/kneaddata"

    setup {
        run("KNEADDATA_DATABASE") {
            script "../../database/main.nf"
            process {
                """
                input[0] = 'human_genome'
                """
            }
        }
    }

    test("fastq.gz") {

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
                            file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_2.fastq.gz", checkIfExists: true)
                        ]
                    ]
                )
                input[1] = KNEADDATA_DATABASE.out.kneaddata_db
                """
            }
        }


        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                    process.out.preprocessed_reads,
                    process.out.versions
                    ).match()
                },
                { assert path(process.out.kneaddata_log[0][1]).text.contains("kneaddata.knead_data - INFO: Running kneaddata v0.12.0") }
            )
        }
    }

    test("fastq.gz - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
                            file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_2.fastq.gz", checkIfExists: true)
                        ]
                    ]
                )
                input[1] = KNEADDATA_DATABASE.out.kneaddata_db
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
