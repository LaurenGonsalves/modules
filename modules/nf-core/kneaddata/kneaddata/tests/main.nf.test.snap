{
    "homo_sapiens - illumina - paired_end_fastq_gz": {
        "content": [
            [
                
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-23T00:21:02.409597642"
    },
    "homo_sapiens - illumina - paired_end_fastq_gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_kneaddata_paired_2.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_kneaddata.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,0dfc969a58fd168cbcd4a9ee87f0c28d"
                ],
                "kneaddata_log": [
                    [
                        {
                            "id": "test"
                        },
                        "test_kneaddata.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "preprocessed_reads": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_kneaddata_paired_2.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0dfc969a58fd168cbcd4a9ee87f0c28d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-23T00:21:13.256817898"
    }
}